core/lib/src/router/route.rs

```diff
@@ -77,17 +77,16 @@ impl Route {
     /// // this is a rank -1 route matching requests to `GET /<name>`
     /// let name = Route::new(Method::Get, "/<name>", handler);
     /// ```
-    pub fn new<S>(m: Method, path: S, handler: Handler) -> Route
+    pub fn new<S>(method: Method, path: S, handler: Handler) -> Route
         where S: AsRef<str>
     {
         let uri = Uri::from(path.as_ref().to_string());
+        let rank = default_rank(&uri);
+
         Route {
+            method, handler, uri, rank,
```

Comment:
This line (and again in `ranked`) now contains more than one of the fields, and they are in a different order. I'm not entirely sold on that.

Reply:
Sergio asked me to do that on the last patch, so I applied it again here.

Reply:
Yeah. This one seems messier, though. Oh well; let's just do this for consistency.